### PR TITLE
[Perf] Implement WindowsMemoryResultParser, WindowsBatteryResultParser and refine WindowsMemoryInspector.ps1

### PR DESCRIPTION
**Pull request by AC-Phoenix**
**Changes**
- Implement WindowsMemoryResultParser, WindowsMemoryMetrics
- Implement WindowsBatteryResultParser, WindowsBatteryMetrics
- Refine WindowsMemoryInspector.ps1

**Next**
- Need to verify if the agent configured with elevated privileges can bypass the UAC popup without changing the UAC configuration.
- Add a new method in ShellUtils to run the admin command if the above one is not feasible.
- Make memory & battery test work with more processes of other Windows apps.

### CODE REVIEW

File: common/src/main/java/com/microsoft/hydralab/performance/parsers/WindowsBatteryResultParser.java [hunk 1]
```diff
@@ -2,14 +2,227 @@
...
+ * 1. To minimize the overhead, you should gather data as infrequently as possible for your measurement. Changing the
+ * sampling rate affects all SRUM providers and should only be done for testing purposes. Changing Tier1Period to < 10
+ * seconds or > 180 seconds will result in the default value of 60 seconds being used.
+ * 2. It is recommended to start parsing the result file 20 seconds after the last power data collection. This is
+ * because even though the execution of the corresponding powershell has done, the results may not have been written to
+ * the hard disk completely because the OS has not yet completed the last step of writing due to big size of the result
+ * file is too large.
+ */
+@Data
+class WindowsBatteryParsedData {
```
Comment (taoran6) [inline common/src/main/java/com/microsoft/hydralab/performance/parsers/WindowsBatteryResultParser.java:38]: nit: I suggest we extract this class to `com.microsoft.hydralab.performance.Entity`. As we may use it for UI visualization later.

File: common/src/main/java/com/microsoft/hydralab/performance/parsers/WindowsBatteryResultParser.java [hunk 2]
```diff
@@ -2,14 +2,227 @@
...
+        windowsBatteryMetrics.setTimeStamp(getOneStringField(fieldValues, "TimeStamp", columnNameToIndexMap));
+        return windowsBatteryMetrics;
+    }
+
+    private long getOneMetric(String[] fieldValues, String metricName, Map<String, Integer>columnNameToIndexMap)
+    {
+        int index = columnNameToIndexMap.getOrDefault(metricName, -1);
+        if (index != -1)
+        {
+            return Long.parseLong(fieldValues[index]);
```
Comment (taoran6) [inline common/src/main/java/com/microsoft/hydralab/performance/parsers/WindowsBatteryResultParser.java:212]: Please make sure parseLong won't throw an exception here if the param is not in a valid float format, I suggest we should handle the exception.

File: common/src/main/java/com/microsoft/hydralab/performance/parsers/WindowsBatteryResultParser.java (hunk 1 above)
Comment (AC-Phoenix) [reply common/src/main/java/com/microsoft/hydralab/performance/parsers/WindowsBatteryResultParser.java]: Cool. They will be extracted to WindowsBatteryParsedData.java and WindowsMemoryParsedData.java

File: common/src/main/java/com/microsoft/hydralab/performance/parsers/WindowsBatteryResultParser.java (hunk 2 above)
Comment (AC-Phoenix) [reply common/src/main/java/com/microsoft/hydralab/performance/parsers/WindowsBatteryResultParser.java]: Thanks, a try-catch added.

**Review by taoran6 — APPROVED** [review]

**Pull request merged**